src/libcollections/string.rs

```diff
@@ -527,9 +595,15 @@ impl String {
         self.vec.capacity()
     }
 
-    /// Reserves capacity for at least `additional` more bytes to be inserted
-    /// in the given `String`. The collection may reserve more space to avoid
-    /// frequent reallocations.
+    /// Increases this `String`'s capacity by at least `additional` bytes.
```

Comment:
I think this needs to be carefully worded as the capacity doesn't actually always increase here. If you have a length 2 string with 10 bytes of capacity, reserve(8) won't actually do anything. This actually just makes sure that `n` more insertions can succeed without reallocating.